frontend/src/components/resourceMap/KubeObjectGlance.tsx

```diff
@@ -0,0 +1,147 @@
+import { Icon } from '@iconify/react';
+import { Box } from '@mui/system';
+import { memo, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { KubeObject } from '../../lib/k8s/cluster';
+import Event from '../../lib/k8s/event';
+import { DateLabel, StatusLabel } from '../common/Label';
+import { makePodStatusLabel } from '../pod/List';
+
+/**
+ * Little Popup preview of a Kube object
+ */
+export const KubeObjectGlance = memo(({ resource }: { resource: KubeObject }) => {
+  const { t } = useTranslation();
+  const [events, setEvents] = useState<Event[]>([]);
+  useEffect(() => {
+    Event.objectEvents(resource).then(it => setEvents(it));
+  }, []);
+
+  const kind = resource.kind;
+
+  const sections = [];
+
+  if (kind === 'Pod') {
```

Comment:
Maybe abstracting each resource component into its own functional component would make the code easier?

Reply:
if this grows larger we can refactor it. for now I think this is simple enough

Reply:
Having separate components here would be better for readability. Also better for inspecting in developer tools, and for adding tests.

Reply:
This is done